plugins/net_plugin/net_plugin.cpp

```diff
@@ -969,15 +1070,31 @@ namespace eosio {
       update_endpoints();
       boost::asio::ip::tcp::no_delay nodelay( true );
       boost::system::error_code ec;
-      socket->set_option( nodelay, ec );
+      socket.get_socket().set_option( nodelay, ec );
       if( ec ) {
          fc_elog( logger, "connection failed (set_option) ${peer}: ${e1}", ("peer", peer_name())( "e1", ec.message() ) );
          close();
          return false;
       } else {
-         fc_dlog( logger, "connected to ${peer}", ("peer", peer_name()) );
-         socket_open = true;
-         start_read_message();
+
+         auto start_read = [this](){
+            fc_dlog( logger, "connected to ${peer}", ("peer", peer_name()) );
+            socket_open = true;
+            start_read_message();
+         };
+         if (socket.use_ssl) {
+            socket.ssl_socket->async_handshake(ssl::stream_base::server,
```

Comment:
Nothing keeps this `ssl_socket` alive for life of async call.

Reply:
done